src/transformers/modeling_flax_utils.py

```diff
@@ -687,7 +723,19 @@ def from_pretrained(
             pretrained_model_name_or_path = str(pretrained_model_name_or_path)
             is_local = os.path.isdir(pretrained_model_name_or_path)
             if os.path.isdir(pretrained_model_name_or_path):
-                if from_pt and os.path.isfile(os.path.join(pretrained_model_name_or_path, subfolder, WEIGHTS_NAME)):
+                if is_safetensors_available() and os.path.isfile(
+                    os.path.join(pretrained_model_name_or_path, SAFE_WEIGHTS_NAME)
+                ):
+                    # Load from a safetensors checkpoint
+                    archive_file = os.path.join(pretrained_model_name_or_path, SAFE_WEIGHTS_NAME)
+                elif is_safetensors_available() and os.path.isfile(
+                    os.path.join(pretrained_model_name_or_path, SAFE_WEIGHTS_INDEX_NAME)
+                ):
+                    # Load from a sharded safetensors checkpoint
+                    archive_file = os.path.join(pretrained_model_name_or_path, SAFE_WEIGHTS_INDEX_NAME)
+                    is_sharded = True
+                    raise NotImplementedError("Support for sharded checkpoints using safetensors is coming soon!")
```

Comment:
If a repository has both:
1. Sharded safetensors weights
2. Sharded Flax weights

Then we currently cannot load the sharded safetensors weights into Flax (which is fine), but we shouldn't let this block the user loading the sharded Flax weights for now?

Should we move the check for sharded safetensors weights to *after* our check for sharded Flax weights? This way we don't block the user loading the latter if available.

Reply:
Very nice catch! That's also the case for TensorFlow, actually. Modifying this and adding tests

Reply:
Addressed in https://github.com/huggingface/transformers/pull/27064/commits/1bd29e03c09284653e1d8ae4a8ed426299b99083